app/controllers/users/email_confirmations_controller.rb

```diff
@@ -42,8 +42,10 @@ def email_address_already_confirmed?
 
     def process_successful_confirmation(email_address)
       confirm_and_notify(email_address)
+      store_from_select_email_flow_in_session
       if current_user
-        flash[:success] = t('devise.confirmations.confirmed')
+        flash[:success] = flash_message_for_successful_signed_in_confirmation
+        session.delete(:from_select_email_flow)
```

Comment:
Do we need this anymore, or can we now expect this to be handled in `AccountsController`, via the assigned session value?

```suggestion
        flash[:success] = t('devise.confirmations.confirmed')
```